lib/motions/general-motions.coffee

```diff
@@ -149,10 +149,16 @@ class MoveLeft extends Motion
 class MoveRight extends Motion
   operatesInclusively: false
 
-  moveCursor: (cursor, count=1) ->
+  # MoveRight can be combined with Delete, which will indicate it by the presence of options.deleting
+  moveCursor: (cursor, count=1, options={}) ->
     _.times count, =>
-      cursor.moveRight() unless cursor.isAtEndOfLine()
-      cursor.moveRight() if settings.wrapLeftRightMotion() and cursor.isAtEndOfLine()
+      goingToNextLine = atom.config.get('vim-mode.wrapLeftRightMotion')
+
+      if not cursor.isAtEndOfLine() # the line is not empty
+        cursor.moveRight()
+        goingToNextLine = false if options.deleting # when deleting, wrapLeftRightMotion only works on an empty line
```

Comment:
Ah, well put. I very much agree that *"we should make Atom feel like vim, but not necessarily reimplement all of vim's inconsistencies."* I've switched to checking for `operator-pending` and all my tests still work (as do all the other tests); the findings above change a bit but not in ways that I mind.